src/main/java/io/confluent/kafka/connect/datagen/DatagenConnector.java

```diff
@@ -107,4 +137,90 @@ private void validateSchemaSource(Config config) {
           .forEach(v -> v.addErrorMessage(SCHEMA_SOURCE_ERR));
     }
   }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  private void validateSchema(Config config) {
+    List<ConfigValue> schemaSources = config.configValues().stream()
+            .filter(v -> DatagenConnectorConfig.isExplicitlySetSchemaSource(v.name(), v.value()))
+            .collect(Collectors.toList());
+
+    Generator.Builder generatorBuilder = new Generator.Builder();
+    // don't perform this validation if multiple schema sources provided
+    if (schemaSources.size() != 1) {
+      return;
+    }
+
+    ConfigValue schemaSource = schemaSources.get(0);
+
+    switch (schemaSource.name()) {
+      case DatagenConnectorConfig.QUICKSTART_CONF:
+        if (!Quickstart.configValues.contains(((String) schemaSource.value()).toLowerCase())) {
+          schemaSource.addErrorMessage(QUICKSTART_ERR);
+          break;
+        }
+        try {
+          generator = generatorBuilder
+                  .schemaStream(getClass().getClassLoader()
+                          .getResourceAsStream(Quickstart.valueOf(((String) schemaSource.value())
+                                  .toUpperCase()).getSchemaFilename()))
+                  .build();
+        } catch (SchemaParseException | IOException e) {
+          schemaSource.addErrorMessage(QUICKSTART_SCHEMA_ERR);
+        }
+        break;
+
+      case DatagenConnectorConfig.SCHEMA_STRING_CONF:
+        try {
+          generator = generatorBuilder.schemaString((String) schemaSource.value()).build();
+        } catch (SchemaParseException e) {
+          schemaSource.addErrorMessage(SCHEMA_STRING_ERR);
+        }
+        break;
```

Comment:
This logic can (and probably should) be implemented as a [ConfigDef.Validator](https://kafka.apache.org/28/javadoc/org/apache/kafka/common/config/ConfigDef.Validator.html) for the schema string property in the config class.